### PR TITLE
add back sol tokens for meld

### PR DESCRIPTION
**Pull request by ruijialin-avalabs**
## Description

**Ticket: [CP-]**

Please provide:

- A summary of the changes
- Motivation and context for this change
- Any dependencies introduced
- (If breaking) migration steps and instructions

- added missing USDC_SOLANA token for meld onramp and offramp
- fix `useTokenIndex` and `useSuppportedCryptoCurrencies` to show the native token correctly

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

- launch buy/withdraw
- you should see SOL and USDC_SOLANA in select a token screen
- Note: on withdraw, you need to have USDC_SOLANA in order to see it in the list

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [X] I have performed a self-review of my code
- [X] I have verified the code works
- [X] I have included screenshots / videos of android and ios
- [X] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/new/features/meld/store.ts [hunk 1]
```diff
@@ -126,12 +133,13 @@ const supportedCryptoCurrenciesStore = create<SupportedCryptoCurrenciesStore>(
         let match: LocalTokenWithBalance | undefined
 
         // Native token
-        if (
-          crypto.chainId &&
-          crypto.contractAddress?.toLowerCase() ===
-            NATIVE_ERC20_TOKEN_CONTRACT_ADDRESS.toLowerCase()
```
[the remark this answers is not in the document]
Comment (ruijialin-avalabs) [reply packages/core-mobile/app/new/features/meld/store.ts:132]: added back L1 tokens here: [c59a37a97d9dac083f6d99c3b6152f2cf229516f](https://github.com/ava-labs/core-mobile/pull/3203/commits/c59a37a97d9dac083f6d99c3b6152f2cf229516f)

**Review by atn4z7 — APPROVED** [review]

**Review by bogdandobritoiu — APPROVED** [review]
[dev tested on iPhone 16 (26) and Pixel 9 (16)](https://app.bitrise.io/app/7d7ca5af7066e290/pipelines/bd90ed95-45b5-44ea-8138-ac116004604a)

Works as expected!
